Foundation/NSPathUtilities.swift

```diff
@@ -10,13 +10,22 @@
 import CoreFoundation
 
 public func NSTemporaryDirectory() -> String {
-    #if os(macOS) || os(iOS)
-    var buf = [Int8](repeating: 0, count: 100)
-    let r = confstr(_CS_DARWIN_USER_TEMP_DIR, &buf, buf.count)
-    if r != 0 && r < buf.count {
-        return String(cString: buf, encoding: .utf8)!
-    }
-    #endif
+#if os(macOS) || os(iOS)
```

Comment:
I’m okay not fighting over this line since the old patch had it still, but it still is incorrect per the guidelines we’ve discussed.

Reply:
Well, I'll still express my disagreement and make the change (I just want to make that change here rather than in #1940 which is for Windows ... this PR has an extra patch that isn't mean to be there).

Reply:
FWIW, I'm not terribly delighted with either solution - but I do think `canImport(Darwin)` is slightly better than a parade of `os()` disjunctions for the time being.